consensus/src/checkpoint.rs

```diff
@@ -0,0 +1,163 @@
+// Copyright 2020 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::common::{Chain, ProtocolStateChain};
+use mina_crypto::base58::Base58Encodable;
+use mina_crypto::hash::{EpochSeed, StateHash};
+use mina_rs_base::protocol_state::ProtocolState;
+#[allow(dead_code)]
+const SLOTS_PER_EPOCH: u32 = 7140;
+
+#[allow(dead_code)]
+fn init_checkpoints(genesis: &mut ProtocolState) {
+    genesis.body.consensus_state.staking_epoch_data.seed = EpochSeed::default();
+    genesis
+        .body
+        .consensus_state
+        .staking_epoch_data
+        .start_checkpoint = StateHash::default();
+    genesis
+        .body
+        .consensus_state
+        .staking_epoch_data
+        .lock_checkpoint = StateHash::default();
+    genesis
+        .body
+        .consensus_state
+        .staking_epoch_data
+        .epoch_length
+        .0 = 1;
+    genesis.body.consensus_state.next_epoch_data.seed =
+        Base58Encodable::from_base58("2vaRh7FQ5wSzmpFReF9gcRKjv48CcJvHs25aqb3SSZiPgHQBy5Dt")
+            .unwrap();
+    genesis
+        .body
+        .consensus_state
+        .next_epoch_data
+        .start_checkpoint = StateHash::default();
+    genesis.body.consensus_state.next_epoch_data.lock_checkpoint =
+        Base58Encodable::from_base58("3NLoKn22eMnyQ7rxh5pxB6vBA3XhSAhhrf7akdqS6HbAKD14Dh1d")
+            .unwrap();
+    genesis.body.consensus_state.next_epoch_data.epoch_length.0 = 2;
+}
+
+#[allow(dead_code)]
+fn is_short_range(c0: &ProtocolStateChain, c1: &ProtocolStateChain) -> bool {
+    if c0.consensus_state().unwrap().epoch_count == c1.consensus_state().unwrap().epoch_count {
+        return c0
+            .consensus_state()
+            .unwrap()
+            .staking_epoch_data
+            .lock_checkpoint
+            == c1
+                .consensus_state()
+                .unwrap()
+                .staking_epoch_data
+                .lock_checkpoint;
+    }
+
+    #[allow(clippy::erasing_op)]
+    if c0.consensus_state().unwrap().epoch_count.0
+        == c1.consensus_state().unwrap().epoch_count.0 + 1
+        && Chain::epoch_slot(c1) >= Some((2 / 3) * SLOTS_PER_EPOCH)
+    {
+        return c0
+            .consensus_state()
+            .unwrap()
+            .staking_epoch_data
+            .lock_checkpoint
+            == c1
+                .consensus_state()
+                .unwrap()
+                .next_epoch_data
+                .lock_checkpoint;
+    } else {
+        return false;
```

Comment:
we can remove `return` from both branches

Reply:
If remove return build will fail.
![Screenshot (305)](https://user-images.githubusercontent.com/8589224/142847979-2ce32bc8-be2f-4466-8a9e-79b684f33ad7.png)